src/components/application_manager/src/app_service_manager.cc

```diff
@@ -72,10 +72,28 @@ smart_objects::SmartObject AppServiceManager::PublishAppService(
 
   std::string service_type = manifest[strings::service_type].asString();
 
-  if (FindServiceByProvider(connection_key, service_type)) {
-    LOG4CXX_WARN(logger_,
-                 "Service already exists for this provider, rejecting");
-    return smart_objects::SmartObject();
+  AppService* found_service =
+      FindServiceByProvider(connection_key, service_type);
+  if (found_service) {
+    LOG4CXX_WARN(logger_, "Service already exists for this provider, updating");
+
+    published_services_lock_.Acquire();
+    found_service->record[strings::service_manifest] = manifest;
```

Comment:
We need to have a duplicate service name check here as well, since the new service name could conflict with another service.

Reply:
I moved the check for the name later in the function to the beginning.